src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -144,6 +144,25 @@ public static boolean isValidAppointmentDelimit(String test, InputSource origin)
         }
     }
 
+    /**
+     * Returns true if the period represented by this {@code Appointment} overlaps with the period
+     * of {@code target} on the same day, inclusive of the {@code start} and {@code end}.<br>
+     *
+     * This method defines an incompatibility relationship between 2 {@code Appointment} objects.
+     */
+    public boolean overlaps(Appointment target) {
```

Comment:
Should this method be renamed to sound more like a Boolean instead as per the Java Coding Standard?

Reply:
This is true - I will rename the method to a more appropriate `hasOverlap` instead.